ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/resources/ResourceOwnerSpec.scala

```diff
@@ -0,0 +1,596 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.resources
+
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{Executors, RejectedExecutionException}
+import java.util.{Timer, TimerTask}
+
+import akka.actor.{Actor, ActorSystem, Props}
+import akka.stream.Materializer
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.{Done, NotUsed}
+import com.digitalasset.platform.resources.ResourceOwnerSpec._
+import org.scalatest.{AsyncWordSpec, Matchers}
+
+import scala.collection.mutable
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+class ResourceOwnerSpec extends AsyncWordSpec with Matchers {
+  "a resource owner" should {
+    "acquire and release a resource" in {
+      val owner = TestResourceOwner(42)
+      owner.hasBeenAcquired should be(false)
```

Comment:
Consider adding `withHint` in tests where you have more than one `should` in a single test case.

Reply:
Adding them. 😄